cmd/kubelet/app/server.go

```diff
@@ -467,6 +467,8 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 		} else {
 			if s.RequireKubeConfig {
 				return fmt.Errorf("invalid kubeconfig: %v", err)
+			} else if (s.KubeConfig.Provided() && !standaloneMode) {
```

Comment:
for your use case, does a log message suffice, or would you have wanted the kubelet to start at all?